pkg/controllers/customeradmin/group_mapping.go

```diff
@@ -0,0 +1,129 @@
+package customeradmin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/openshift/api/user/v1"
+	"github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func setIfDifferent(field *string, value string) bool {
+	changed := (*field != value)
+	*field = value
+	return changed
+}
+
+func setLabelIfDifferent(labels map[string]string, key string, value string) bool {
+	current, exists := labels[key]
+	labels[key] = value
+	return !exists || current != value
+}
+
+// fromMSGraphGroup syncs the values from the given aad group into kubeGroup
+// if kubeGroup is nil, it creates a new one
+// if group is nil, it sets the membership to empty (this is the case if there is no aad group defined)
+// labels are set so one can see the origins of the kubeGroup
+// it returns a boolean "changed" that tells the call if an update is required
+func fromMSGraphGroup(log *logrus.Entry, kubeGroup *v1.Group, group *graphrbac.ADGroup, groupName string, msGroupMembers []graphrbac.AADObject) (*v1.Group, bool) {
+	changed := false
```

Comment:
this changed tracking feels rather involved.  Can't you always construct the new Group, then compare it with the existing one and only call update if you decide you need to?

Reply:
can do